### DISCUSSION TITLE
Host NuGet Package at Enterprise Scope, Available for Organizations managed under said Enterprise

### DISCUSSION THREAD

**Discussion by TerumiMcIntyre** [community/community#77815] [Enterprise]
### Select Topic Area

Question

### Body

Hi there, 

Is it possible to upload/host a private package at the Enterprise Cloud level & have it available to use for all my Organizations that I manage under said Enterprise Cloud account?

**Comment by dakshithadissanayaka** [comment]
Yes, it is possible to upload/host a private package at the Enterprise Cloud level and make it available to use for all your Organizations that you manage under said Enterprise Cloud account. This can be done using GitHub Packages, which is a package hosting service that is included with GitHub Enterprise Cloud.

To upload a private package to GitHub Packages, you will need to follow these steps:

1. Create a new repository in your Enterprise Cloud account.
2. Go to the **Packages** tab for your repository.
3. Click the **Upload package** button.
4. Select the package file that you want to upload.
5. Click the **Upload** button.

Once the package has been uploaded, it will be available to use for all of your Organizations in your Enterprise Cloud account. To use the package, you will need to add it to your repository's dependencies. You can do this by adding the following to your repository's `package.json` file:

```json
{
  "dependencies": {
    "your-package": "https://your-enterprise-cloud-url/organizations/your-organization/packages/your-package"
  }
}
```

Once you have added the package to your repository's dependencies, you can use it in your code.

Here is an example of how to use a private package in your code:

```javascript
const myPackage = require('your-package');

console.log(myPackage.helloWorld());
```

This will print the following output to the console:

```
Hello, world!
```

I hope this helps!

**Comment by Jonnabro** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> Hi there, 
> 
> Is it possible to upload/host a private package at the Enterprise Cloud level & have it available to use for all my Organizations that I manage under said Enterprise Cloud account?

**Comment by tinaheidinger** [comment]
Yes, you can do that! See https://docs.github.com/en/packages/learn-github-packages/configuring-a-packages-access-control-and-visibility#package-creation-visibility-for-organization-members for how to publish an _internal_ package
